Drive/rover_drive_system.hpp

```diff
@@ -22,10 +22,12 @@ class RoverDriveSystem
   /// Mission controls possible input values
   struct MissionControlData
   {
-    bool is_operational;
+    int is_operational;
     char drive_mode;
     float rotation_angle;
     float speed;
+    char * response;
+    char * request;
```

Comment:
This area looks dubious and dangerous. I feel like you are on your way to accessing invalid data with this here.

Reply:
I ended up replacing with char var[300] but I'm not sure if that was right choice. I think std::string would make the most sense however that would mean parsing the GET response parsers would need to be redone... Curious what your thoughts are?